### PR TITLE
Fix hyper

### PR DESCRIPTION
**Pull request by polachok**
Upgrades hyper to latest master

### CODE REVIEW

File: frameworks/Rust/hyper/src/main.rs [hunk 1]
```diff
@@ -33,8 +32,9 @@ impl Service for TechEmpower {
                 Response::new()
                     .with_header(ContentLength(HELLOWORLD.len() as u64))
                     .with_header(ContentType(Mime(TopLevel::Text, SubLevel::Plain, vec![(Attr::Charset, Value::Utf8)])))
+                    .with_header(header::Server("hyper/async".to_owned()))
```
Comment (steveklabnik) [inline frameworks/Rust/hyper/src/main.rs]: standard rust style is `to_string` not `to_owned` (but they do the exact same thing)

**Comment by alexcrichton** [comment]
Oh one strategy this may wish to take as well is to check in `Cargo.lock` and avoid the need for `rev = "..."` in the dependency specification, but that's completely optional!

**Comment by steveklabnik** [comment]
@alexcrichton I don't remember why I didn't do it that way, but I think there was some kind of issue. Can't remember; that was like two years ago or something. I could have messed it up too.

**Comment by alexcrichton** [comment]
Ah yeah [tokio-minihttp](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Rust/tokio-minihttp/Cargo.lock) nowdays has `Cargo.lock`, so I figured it'd be ok to throw in here as well (but again, just an optional extension)

File: frameworks/Rust/hyper/src/main.rs (hunk 1 above)
Comment (polachok) [inline frameworks/Rust/hyper/src/main.rs]: fixed

**Comment by arthurprs** [comment]
It's a bit naive to allocate a string on every reques just to set the server header. I'm sure it doesn't matter in practice but it may show in a hello-world benchmark

**Comment by polachok** [comment]
@arthurprs well, that's how Server is defined https://hyper.rs/hyper/0.8.0/hyper/header/struct.Server.html

**Comment by arthurprs** [comment]
I see, `Server` is also required by the test rules.

**Comment by nbrady-techempower** [comment]
Looks like there's agreement to close this in favor of #2619 

Closing this and merging that. Thanks guys!

**Pull request closed without merging**